### DISCUSSION TITLE
Wrong sun_risetime & sun_settime?

### DISCUSSION THREAD

**Discussion by nocopy074** [tbnobody/OpenDTU#953] [General]
Hi guys, 

first of all thank you for the great work with this project! I have a question about the calculation of the sun_risetime and sun_settime. As far as I understand, it is calculated with a c++ libary. The problem is, that I think it is not calculated correctly. For my geo posistion it calculates this times: 
sun_risetime | "Wednesday, May 24 2023 03:54:00"
sun_settime | "Wednesday, May 24 2023 23:12:00"

But I think for a place in western Germany this times are wrong? Can you help me understanding please?

**Comment by tbnobody** [comment]
Currently the nautical sunset/sunrise is used:
https://github.com/tbnobody/OpenDTU/blob/889e191589726ea857a1173f82671063e8c9568d/src/SunPosition.cpp#L48C13-L49

See e.g. https://detlefhahn.de/segeln/astro/zeitgleichung.php

**Reply by nocopy074** [reply]
Is there a reason why? And could I change it to SUNSET_OFFICIAL ?

Edit: I am asking this, because I have the problem, that the dtu tries to communincate to my inverters until the calculated sunset is reached, but they can not answer at all, because there is not enough sunlight for them to work.

**Reply by traeu** [reply]
@nocopy074 just curious: is there a reason why you want to stop polling data at night? I enabled "poll data at night" because I thought it won't hurt if the DTU always tries to get data and if there is an answer, fine, and if the inverter doesn't answer, also fine.
Now with this discussion about adjusting sunrise/set times I start to wonder why people want to predict when communication is possible and when not, instead of just always trying.

**Reply by nocopy074** [reply]
I am using an fork of this project called openDTUonBattery and I could observe a misbehavior during the night while trying to poll 2 inverter not answering and controlling one inverter on a battery. This misbehavior was always gone when the calculated night time was reached. So I was looking at the code and saw that calculation and was wondering about this. I think for me it is better if the calculated time is more near my real night time for the inverters. I think its also better for the energie consumption, but here I am not shure whether it really has an influence on it.

**Reply by traeu** [reply]
thanks for your answer! looks like I don't need to worry about trying to poll data all night when I only use openDTU for the simple task of collecting metrics from one single inverter and don't observe any strange behavior.
I also thought about energy consumption, but even the version of NRF24L01 with on-board amplifier and 20 dBm output consumes max 115 mA, nothing to really worry about...

**Comment by homeautomation2022** [comment]
Ich würde das nautical sehr gern in civil geändert haben, das müssen aber die Ost/Westler nächsten Monat bestätigen, dass das reicht bzw. muss man beobachten, ob es da schon dunkel ist.
Bei mir mit Süd Ausrichtung ist Civil immernoch genug übertrieben

**Reply by tbnobody** [reply]
Bekommst demnächst einen Parameter in den Einstellungen :-)

**Reply by NintendoTito** [reply]
Ehrlich gesagt finde ich de Einstellung gar nicht... Wo finde ich die Möglichkeit die DTU nachts in den DeepSleep zu versetzen?
